File: Mixter/Domain/Subscriptions/Subscription.cs
namespace Mixter.Domain.Subscriptions
{
    public class Subscription
    {
        public static void FollowUser(IEventPublisher eventPublisher, UserId follower, UserId followee)
        {
            eventPublisher.Publish(new UserFollowed(new SubscriptionId(follower, followee)));
        }
    }
}

File: src/main/java/mixter/Subscription.java
package mixter;

public class Subscription {
    public static void follow(UserId follower, UserId followee, EventPublisher eventPublisher) {
        UserFollowed userFollowed = new UserFollowed(new SubscriptionId(follower, followee));
        eventPublisher.publish(userFollowed);
    }
}
